File: sdks/node/src/api/template/req.ts
import type { AnswerReq } from '../core/answer_req.js';

interface TemplateValidateReq {
  answers: Record<string, AnswerReq>;
  deterministicStates: Record<string, string>;
  validate: string;
}

interface TemplateAnswerReq {
  answers: Record<string, AnswerReq>;
  deterministicStates: Record<string, string>;
}

// export all
export type { TemplateAnswerReq, TemplateValidateReq };

File: sdks/python/cyan_sdk/api/template/req.py
from typing import Dict, List

from cyan_sdk.api.base_model import CyanBaseModel

from cyan_sdk.api.core.answer_req import AnswerReq


class TemplateValidateReq(CyanBaseModel):
    answers: List[AnswerReq]
    deterministic_states: List[Dict[str, str]]
    validate: str


class TemplateAnswerReq(CyanBaseModel):
    answers: List[AnswerReq]
    deterministic_states: List[Dict[str, str]]
